internal/node/reverse_proxy/subscriber.go

```diff
@@ -47,3 +66,36 @@ func (e *ProcessProxyRulesExecutor) Execute(update hdb.StateUpdate) error {
 func (e *ProcessProxyRulesExecutor) PostHook(update hdb.StateUpdate) error {
 	return nil
 }
+
+type AddProxyRulesExecutor struct {
+	RuleSet *RuleSet
+}
+
+func (e *AddProxyRulesExecutor) TransitionType() string {
+	return node.TransitionAddReverseProxyRule
+}
+
+func (e *AddProxyRulesExecutor) ShouldExecute(update hdb.StateUpdate) (bool, error) {
+	// This process is lightweight, so we can execute it every time
```

Comment:
hmm, isn't this bad practice though? like if we ever wanna test what executed, it doesn't make sense for this to execute on everything. plus seems like we may have many of these in the future

Reply:
You're right + this won't exactly be lightweight if we have multiple nodes talking over raft.

Reply:
I can reword this, there isn't exactly much to check here. Any conflicts should have been handled by the validation step in the state machine.